partner_risk_insurance/models/account_move.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2024 Moduon Team S.L. <info@moduon.team>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, exceptions, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    insured_with_credit_policy = fields.Boolean(
+        string="Insured by a Credit Policy",
+        compute="_compute_insured_with_credit_policy",
+        store=True,
+        readonly=False,
+        help="Indicates if the invoice is insured by a credit policy.",
+    )
+    credit_policy_company_id = fields.Many2one(
+        string="Insurance Company",
+        comodel_name="credit.policy.company",
+        compute="_compute_credit_policy",
+        store=True,
+        readonly=True,
+        ondelete="restrict",
+        help="The insurance company that insures the invoices of the partner.",
+    )
+    credit_policy_state_id = fields.Many2one(
+        string="Insurance Policy",
+        comodel_name="credit.policy.state",
+        compute="_compute_credit_policy",
+        store=True,
+        readonly=True,
+        ondelete="restrict",
+        domain="[('insure_invoices', '=', True)]",
+        help="The insurance policy that insures the invoices of the partner.",
+    )
+
+    @api.depends("partner_id", "move_type")
+    def _compute_insured_with_credit_policy(self):
+        """Compute the insured_with_credit_policy field when partner changes."""
+        self.update(
+            {
+                "insured_with_credit_policy": False,
+                "credit_policy_state_id": False,
+                "credit_policy_company_id": False,
+            }
+        )
+        for record in self:
+            if record.move_type != "out_invoice":
+                continue
+            partner = record.partner_id.commercial_partner_id
+            if partner.credit_policy_insure_invoices:
+                record.insured_with_credit_policy = True
+                record.credit_policy_company_id = partner.credit_policy_company_id
+                record.credit_policy_state_id = partner.credit_policy_state_id
+
+    @api.depends("insured_with_credit_policy")
+    def _compute_credit_policy(self):
+        """Compute if the invoice is insured by a credit policy if
+        `credit_policy_insure_invoices` field changes."""
+        self.update(
+            {
+                "credit_policy_state_id": False,
+                "credit_policy_company_id": False,
+            }
+        )
+        for record in self:
+            if record.move_type != "out_invoice":
+                continue
+            partner = record.partner_id.commercial_partner_id
+            if record.insured_with_credit_policy:
```

Comment:
**suggestion:** I see a lot of code duplicated among the 2 compute methods, and one depends on the other, so... would it make more sense to just use one single compute method for all 3 fields?

Reply:
I can't do the same behavior (check and uncheck the `insured_with_credit_policy` field) if I put all the logic in the same function.
I'm not able to respect the decision to insure or not insure an invoice on the creation of the invoice. 
This is why I've divided the logic in 2 functions.

`_compute_insured_with_credit_policy` function only sets the check, and the `_compute_credit_policy` gets the data from the partner based on the state of the check.
If all goes into the same function, with `if record.insured_with_credit_policy` and `if partner.credit_policy_insure_invoices`, I can't know if it's a computation based on the partner or the check.